timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -484,6 +491,37 @@ public void stressTestForPaxosEndpoints() {
         }
     }
 
+    @Test
+    public void leaderCanProcessMultiClientLeaderTimeRequest() {
+        TestableTimelockServer leader = cluster.currentLeaderFor(client.namespace());
+        MultiClientConjureTimelockService multiClientConjureTimelockService = leader.multiClientService();
+        Set<String> expectedNamespaces = ImmutableSet.of("alpha", "beta");
+        List<NamespacedLeaderTime> leaderTimes =
+                multiClientConjureTimelockService.leaderTimes(AUTH_HEADER, expectedNamespaces);
+        Set<String> namespaces =
+                leaderTimes.stream().map(NamespacedLeaderTime::getNamespace).collect(Collectors.toSet());
+        assertThat(namespaces).hasSameElementsAs(expectedNamespaces);
+
+        Set<LeadershipId> leadershipIds = leaderTimes.stream()
+                .map(NamespacedLeaderTime::getLeaderTime)
+                .map(LeaderTime::id)
+                .collect(Collectors.toSet());
+        assertThat(leadershipIds).hasSameSizeAs(expectedNamespaces);
```

Comment:
maybe a bit overkill: if we run these against the regular conjuretimelockservice.getLeaderTime we get the same answer right?

Reply:
Same leader will serve the request, but leaderTime will be different (time elapsed b/w two requests). I will add this test for commitTs request as well, where I can add stricter assertions on the response.

Reply:
Yeah, makes sense. I think I intended an ID comparison, which you've added below.